vcr/credential/resolver.go

```diff
@@ -59,17 +60,23 @@ func ExtractTypes(credential vc.VerifiableCredential) []string {
 
 // PresentationSigner returns the DID of the signer of the presentation.
 // It does not do any signature validation.
-// For JWTs it returns the issuer (iss) of the JWT.
+// For JWTs it returns the did in the kid header of the JWT.
 // For JSON-LD it returns the verification method of the proof.
 func PresentationSigner(presentation vc.VerifiablePresentation) (*did.DID, error) {
 	switch presentation.Format() {
 	case vc.JWTPresentationProofFormat:
-		token := presentation.JWT()
-		issuer := token.Issuer()
-		if issuer == "" {
-			return nil, errors.New("JWT presentation does not have 'iss' claim")
+		kid, _, err := crypto.JWTKidAlg(presentation.Raw())
+		if err != nil {
+			return nil, err
+		}
+		if kid == "" {
+			return nil, errors.New("no kid header in JWT")
+		}
```

Comment:
`verifier.resolveSigningKey` tries to apply some magic in case the kid header is missing or only contains the fragment. Should this also be applied here?

Reply:
I'd say not if missing, but yes if it only contains the fragment?

Reply:
deferred to #2684